test/tests/functional/pbs_test_job_requeue_delay.py

```diff
@@ -0,0 +1,94 @@
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under a
+# commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestJobRequeueDelayAttribute(TestFunctional):
+    """
+    This test suite is for testing the new job_requeue_delay attribute
+    that replaced job_requeue_timeout.
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        if len(self.moms) != 2:
+            self.logger.error('test requires two MoMs as input, ' +
+                              '  use -p moms=<mom1>:<mom2>')
+            self.assertEqual(len(self.moms), 2)
+
+        # PBSTestSuite returns the moms passed in as parameters as dictionary
+        # of hostname and MoM object
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.momA.delete_vnode_defs()
+        self.momB.delete_vnode_defs()
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_SET, SERVER, {
+                                 'job_requeue_delay': 1}, expect=True)
+        self.assertTrue(rc)
+
+    def test_error_message(self):
+        j = Job(TEST_USER, attrs={ATTR_N: 'job_requeue_delay'})
+
+        test = []
+        test += ['dd if=/dev/zero of=file bs=1024 count=0 seek=10240\n']
+        test += ['cat file\n']
+        test += ['sleep 30\n']
+
+        j.create_script(test, hostname=self.server.client)
+        jid = self.server.submit(j)
```

Comment:
Well, this test case is dependent on the machine it is running. With excellent network speed and good configurations there is no assurance that error message will be thrown, And in case it isn't then the test case will show up as passed.
But I don't see a way out too. So, probably what you have done is the best that we can do for now.